core/src/components/button/button.ionic.scss

```diff
@@ -123,45 +122,33 @@
   }
 }
 
-/* Button extra small */
-:host(.button-has-icon-only.button-xsmall) {
-  --padding-end: #{$button-has-icon-only-padding-end-xsmall};
-}
-
-/* Button small */
-:host(.button-has-icon-only.button-small) {
-  --padding-end: #{$button-has-icon-only-padding-end-small};
-}
-
-/* Default */
 :host(.button-has-icon-only) {
-  --padding-end: #{$button-has-icon-only-padding-end};
-}
-
-/* Button large */
-:host(.button-has-icon-only.button-large) {
-  --padding-end: #{$button-has-icon-only-padding-end-large};
-}
+  --padding-top: 0;
+  --padding-bottom: var(--padding-top);
+  --padding-end: var(--padding-top);
+  --padding-start: var(--padding-end);
```

Comment:
It seems like it isn't needed, I think I had it there in case we changed the `--padding-top` in the future, but the screenshot tests will catch that so I removed it: https://github.com/ionic-team/ionic-framework/pull/29404/commits/075f552f7c2bcc9ad4fd5b096d8ab2387a419dfd

I also completely forgot to add _why_ I made this change to the PR description so I just updated that to include a section on my changes for the icon only buttons.